### PR TITLE
REP-90 Script to import businesses for REW

### PR DESCRIPTION
**Pull request by edwh**
Adds script to import from spreadsheet.

Also adds part of REP-96, so that we will display businesses with different criteria, otherwise it's harder to test this.  To achieve that we now pass the region to the server, and then user per-region criteria.

### CODE REVIEW

File: app/Console/Commands/ImportBusinessesSpreadsheet.php [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                    } else {
+                        $uniques[$key] = TRUE;
+
+                        // Tweak the URL.
+                        if ($website) {
+                            if (stripos($website, 'http') === FALSE) {
+                                $website = "https://$website";
+                            }
+
+                            $website = preg_replace('/\/#$/', '', $website);
```
Comment (ngm) [inline app/Console/Commands/ImportBusinessesSpreadsheet.php:121]: Are these removing trailing # and /?  Could you put a comment as to what these are doing, it isn't immediately evident (perhaps my regex isn't up to scratch!)

File: app/Console/Commands/ImportBusinessesSpreadsheet.php [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+                        $business->setUpdatedAt(new \DateTime("now"));
+
+                        $point = null;
+
+                        try {
+                            $point = $geocoder->geocode("$address,$city,$borough,$postcode");
+
+                            if ($point && ($point->getLatitude() < $swlat || $point->getLatitude() > $nelat || $point->getLongitude() < $swlng || $point->getLongitude() > $nelng)) {
+                                $this->error("$name address $address geocodes to invalid lat/lng " . $point->getLatitude() . "," . $point->getLongitude());
+                                $point = null;
```
Comment (ngm) [inline app/Console/Commands/ImportBusinessesSpreadsheet.php:155]: To check, are we saying here that if the address doesn't geocode, we are ignoring the business and logging it as a problem?  Presumably to then send to REW and ask them to fix.

**Review by ngm — COMMENTED** [review]
Great stuff, just some minor (hopefully not too pernickety) comments.

File: app/Console/Commands/ImportBusinessesSpreadsheet.php [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+     * @var string
+     */
+    protected $signature = 'restart:import:spreadsheet {file}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Command description';
```
[the remark this answers is not in the document]
Comment (edwh) [reply app/Console/Commands/ImportBusinessesSpreadsheet.php:36]: Erm, yes.

File: app/Console/Commands/ImportBusinessesSpreadsheet.php (hunk 1 above)
Comment (edwh) [reply app/Console/Commands/ImportBusinessesSpreadsheet.php:121]: Will comment.

File: app/Console/Commands/ImportBusinessesSpreadsheet.php (hunk 2 above)
Comment (edwh) [reply app/Console/Commands/ImportBusinessesSpreadsheet.php:155]: Yes; will comment to that effect.

**Pull request merged**